ynput/OpenPype
# Kitsu: note family requirements

## Brief description
Allowing to add family requirements to `IntegrateKitsuNote` task status change.

## Description
Adds a `Family requirements` setting to `Integrate Kitsu Note`, so you can add requirements to determine if kitsu task status should be changed based on which families are published or not. For instance you could have the status change only if another subset than workfile is published (but workfile can still be included) by adding an item set to `Not equal` and `workfile`.

## Additional info
When adding an item to this setting, if `Equality` is set to `Equal`, task status will be updated if any published subset is in the indicated family. If it is set to `Not equal`, task status will be updated if any published subset is in another family than the one indicated. You can add as many suggestions as you need in this way.

## Testing notes
- In `Project settings -> Kitsu -> Integrate Kitsu Note`, set `Set status on note` to True, `Note shortname` to the status shorname you want, and set `Family requirements` the way you need. If you `Status exceptions`, make sure you do not trigger the exceptions you set to perform the next series of tests (Except the last one).
- Publish an asset in a way that would NOT fulfill the family requirements you set, and check that your task status has not been updated.
- Publish an asset in a way that would fulfill the family requirements you set, and check that your task status has been updated.
- Now you can also publish while triggering a Status exception and check that you task status has not been updated.
